pkg/controller/constraint/constraint_controller.go

```diff
@@ -47,72 +47,81 @@ var (
 )
 
 const (
-	finalizerName = "finalizers.gatekeeper.sh/constraint"
+	finalizerName      = "finalizers.gatekeeper.sh/constraint"
+	constraintsGroup   = "constraints.gatekeeper.sh"
+	constraintsVersion = "v1beta1"
 )
 
 type Adder struct {
 	Opa              *opa.Client
 	ConstraintsCache *ConstraintsCache
+	WatchManager     *watch.Manager
+	Events           <-chan event.GenericEvent
 }
 
-type ConstraintsCache struct {
-	mux   sync.RWMutex
-	cache map[string]tags
+func (a *Adder) InjectOpa(o *opa.Client) {
+	a.Opa = o
 }
 
-type tags struct {
-	enforcementAction util.EnforcementAction
-	status            metrics.Status
+func (a *Adder) InjectWatchManager(w *watch.Manager) {
+	a.WatchManager = w
 }
 
 // Add creates a new Constraint Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
-func (a *Adder) Add(mgr manager.Manager, gvk schema.GroupVersionKind, cs *watch.ControllerSwitch) error {
+func (a *Adder) Add(mgr manager.Manager) error {
 	reporter, err := newStatsReporter()
 	if err != nil {
 		log.Error(err, "StatsReporter could not start")
 		return err
 	}
 
-	r := newReconciler(mgr, gvk, a.Opa, cs, reporter, a.ConstraintsCache)
```

Comment:
controller switch will remain necessary until we can get rid of the garbage collection for byPod status (controller must be stopped to avoid contention with cleaning loop).

Reply:
Got it - added controller switch back in the latest commit.